substrate/runtime/primitives/src/traits.rs

```diff
@@ -40,15 +40,25 @@ pub trait Verify {
 	fn verify<L: Lazy<[u8]>>(&self, msg: L, signer: &Self::Signer) -> bool;
 }
 
+/// Means of changing one type into another in a manner dependent on the source type.
+pub trait AuxLookup {
+	/// Type to lookup from.
+	type Source;
+	/// Type to lookup into.
+	type Target;
+	/// Attempt a lookup.
+	fn lookup(s: Self::Source) -> result::Result<Self::Target, &'static str>;
```

Comment:
I don't think this should be implicitly global. taking `&self` is strictly more powerful because it's easy to us a ZST that uses globals internally.

Reply:
that means passing through an instance, which would be difficult to make work with the rest of the runtime code. perhaps another PR, though it should probably be delayed until/unless we actually need it, and if so i'd probably take a much deeper look at how to rework the runtime.